File: 3-kotlin/ktor-pkuszn-bot/src/main/kotlin/com/example/Purchase.kt
package com.example

import kotlinx.serialization.Serializable

@Serializable
public data class Purchase(val id: Int, val idProduct: Int, val idUser: Int, val price: Float, val quantity: Int, val purchaseDate: String, val deliveryType: Int, val paymentType: Int) {
    fun toDiscordMessage(): String {
        return """
             Purchase ID: $id
             Product ID: $idProduct
             User ID: $idUser
             Price: $price
             Quantity: $quantity
             Purchase Date: $purchaseDate
             Delivery Type: $deliveryType
             Payment Type: $paymentType
             """.trimIndent()
    }
}

File: 5-react/my-app/src/models/purchase.js
export class Purchase {
    idProduct;
    idUser;
    price;
    quantity;
    purchaseDate;
    deliveryType;
    paymentType;

    constructor(
        idProduct,
        idUser,
        price,
        quantity,
        purchaseDate,
        deliveryType,
        paymentType
    ) {
        this.idProduct = idProduct;
        this.idUser = idUser;
        this.price = price;
        this.quantity = quantity;
        this.purchaseDate = purchaseDate;
        this.deliveryType = deliveryType;
        this.paymentType = paymentType;
    }
}